server/monitor.go

```diff
@@ -2686,9 +2688,98 @@ func (s *Server) HandleJsz(w http.ResponseWriter, r *http.Request) {
 	}
 	b, err := json.MarshalIndent(l, "", "  ")
 	if err != nil {
-		s.Errorf("Error marshaling response to /leafz request: %v", err)
+		s.Errorf("Error marshaling response to /jsz request: %v", err)
 	}
 
 	// Handle response
 	ResponseHandler(w, r, b)
 }
+
+type HealthStatus struct {
+	Status string `json:"status"`
+	Error  string `json:"error,omitempty"`
+}
+
+// https://tools.ietf.org/id/draft-inadarei-api-health-check-05.html
+func (s *Server) HandleHealthz(w http.ResponseWriter, r *http.Request) {
+	s.mu.Lock()
+	s.httpReqStats[HealthzPath]++
+	s.mu.Unlock()
+
+	var health = &HealthStatus{Status: "ok"}
+
+	if err := s.readyForConnections(time.Millisecond); err != nil {
+		health.Status = "error"
+		health.Error = err.Error()
+		w.WriteHeader(http.StatusServiceUnavailable)
+	} else if js := s.getJetStream(); js != nil {
+		// Check JetStream status here.
+		js.mu.RLock()
+		clustered, cc := !js.standAlone, js.cluster
+		js.mu.RUnlock()
+		if clustered {
+			// We do more checking for clustered mode to allow for proper rolling updates.
+			// We will make sure that we have seen the meta leader and that we are current with all assets.
+			node := js.getMetaGroup()
```

Comment:
I think this part is very useful for when the container is starting up (before being considered to be able to accept connections from clients // startup probe), but for the liveness probe that checks periodically that the server is healthy after starting successfully we may have to skip this part otherwise the server may get restarted by not being caught up after running for awhile?

Maybe we could define some special http headers for the endpoint with the preconditions that we want to assert, during startup we want to ensure that JS is caught up so can make the k8s probe send a header like `Check-JS-Cluster-Current: true`, but then during liveness check it is only needed to check that JetStream has not been auto disabled so then a different header is set like `Check-JS-Enabled: true`.

Reply:
This is more for startup for sure. For liveness, how should we define when we would ever want k8s to stop the server? I am struggling to think of one, so for liveness its varz maybe and if it returns a 200?

Reply:
For liveness, in case JS is enabled and it becomes auto disabled due to an I/O error, instead of leaving the server running without JS then maybe make the healthcheck fail and let it be restarted to try to auto-recover? For non JS deployments, just checking `/`/ or `/varz` as we are right now would suffice.